src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -191,6 +192,34 @@ CitusCustomScanPathPlan(PlannerInfo *root,
 	 */
 	citusPath->remoteScan->scan.plan.targetlist = tlist;
 
+	/*
+	 * The custom_scan_tlist contains target entries for to the "output" of the call
+	 * to citus_extradata_container, which is actually replaced by a CustomScan.
+	 * The target entries are initialized with varno 1 (see MasterTargetList), since
+	 * it's currently the only relation in the join tree of the masterQuery.
+	 *
+	 * If the citus_extradata_container function call is not the first relation to
+	 * appear in the flattened rtable for the entire plan, then varno is now pointing
+	 * to the wrong relation and needs to be updated.
+	 *
+	 * An example is when the masterQuery is INSERT INTO local SELECT .. FROM
```

Comment:
I had trouble reading and understanding this part. 

```suggestion
	 * An example:
	 * INSERT INTO local_table SELECT .. FROM citus_extradata_container
	 * In that case the varno of citusdata_extradata_container
	 * should be 3, because it is preceded by range table entries for
	 * local_table and the subquery.
```

Reply:
It is important to state it is the masterQuery. So I made something in between.